aiida/cmdline/params/options/commands/code.py

```diff
@@ -137,6 +137,15 @@ def validate_label_uniqueness(ctx, _, value):
     help='Name of the computer, on which the code is installed.'
 )
 
+WITH_MPI = OverridableOption(
+    '--with-mpi/--without-mpi',
+    is_eager=False,
+    default=True,
```

Comment:
```suggestion
    default=True,
    required_fn=is_on_computer,
    prompt_fn=is_on_computer,
```
I think it is more make sense to only set this for the remote code. Compiling with mpi usually link to the library therefore for the local code we should always run in serial. Also check https://www.open-mpi.org/faq/?category=mpi-apps#static-mpi-apps . It says 'Can I build 100% static MPI applications? Fully static linking is not for the weak, and it is not recommended. But it is possible, with some caveats.'

Reply:
It is actually possible to run python scripts with mpirun: https://rabernat.github.io/research_computing/parallel-programming-with-mpi-for-python.html